services/facility_queue/facility_queue.py

```diff
@@ -56,63 +55,74 @@ async def load_from_db(self):
     async def service(self):
         while True:
             req = await queue.get()
-            print(f"{req.method} request to [{req.endpoint}]")
 
-            session = Session()
-            followup_request = session.query(FollowupRequest).get(
-                req.followup_request_id
-            )
-            instrument = followup_request.allocation.instrument
-            altdata = followup_request.allocation.altdata
-
-            response = request_session.request(
-                req.method,
-                req.endpoint,
-                json=req.data,
-                params=req.params,
-                headers=req.headers,
-            )
+            print(f"Executing request {req.id}")
 
-            if instrument.name == "ATLAS":
-                from skyportal.facility_apis.atlas import commit_photometry
-
-                if response.status_code == 200:
-                    try:
-                        json_response = response.json()
-                    except Exception:
-                        raise ('No JSON data returned in request')
-
-                    if json_response['finishtimestamp']:
-                        req.status = "Committing photometry to database"
-                        commit_photometry(
-                            json_response,
-                            altdata,
-                            followup_request.id,
-                            instrument.id,
-                            followup_request.requester.id,
-                        )
-                        req.status = 'complete'
-                        session.add(req)
-
-                    elif json_response['starttimestamp']:
-                        followup_request.status = f"Task is running (started at {json_response['starttimestamp']})"
-                        await self.put(req)
-                    else:
-                        followup_request.status = f"Waiting for job to start (queued at {json_response['timestamp']})"
-                        await self.put(req)
-                else:
-                    followup_request.status = f'error: {response.content}'
+            dt = datetime.utcnow() - req.last_query
+            if dt < WAIT_TIME_BETWEEN_QUERIES:
+                print(
+                    f"Holding {req.id} for {(WAIT_TIME_BETWEEN_QUERIES-dt).seconds} seconds"
+                )
+                await self.put(req)
+            else:
+                session = Session()
```

Comment:
with Session syntax?

Reply:
This gives me issues like:

ERROR:tornado.application:Exception in callback functools.partial(<bound method IOLoop._discard_future_result of <tornado.platform.asyncio.AsyncIOMainLoop object at 0x15e06c040>>, <Task finished name='Task-2' coro=<FacilityQueue.service() done, defined at /Users/mcoughlin/Code/ZTF/skyportal-branches/query_queue/skyportal/services/facility_queue/facility_queue.py:55> exception=DetachedInstanceError('Instance <FacilityTransactionRequest at 0x15e075b50> is not bound to a Session; attribute refresh operation cannot proceed')>)

Reply:
I reworked the sessions to make this happen.